src/Mvc/src/Microsoft.AspNetCore.Mvc.ViewFeatures/HtmlHelperComponentExtensions.cs

```diff
@@ -46,39 +46,28 @@ public static class HtmlHelperComponentExtensions
         {
             if (htmlHelper == null)
             {
-                throw new System.ArgumentNullException(nameof(htmlHelper));
+                throw new ArgumentNullException(nameof(htmlHelper));
             }
 
-            var serviceProvider = htmlHelper.ViewContext.HttpContext.RequestServices;
-            var encoder = serviceProvider.GetRequiredService<HtmlEncoder>();
+            var httpContext = htmlHelper.ViewContext.HttpContext;
+            var serviceProvider = httpContext.RequestServices;
+            var prerrenderContext = serviceProvider.GetRequiredService<MvcPrerrenderingContext>();
```

Comment:
MvcPrerrenderingContext This is just a container to get all the services in one go.